.github/workflows/ci.yml

```diff
@@ -37,6 +37,7 @@ jobs:
          environment-name: geocat-examples
          environment-file: conda_environment.yml
          cache-environment: true
+         cache-environment-key: "${{runner.os}}-${{runner.arch}}-py${{matrix.python-version}}-${{env.TODAY}}"
```

Comment:
It doesn't look like all the info in the cache key is making it to the cache tag.

This is a cache from this brach:
<img width="633" alt="image" src="https://github.com/NCAR/geocat-examples/assets/38434768/95078906-f57f-4ac8-9e37-cdc3c5a6bbc9">

A cache from xarray using a similar key format looks like this:
<img width="632" alt="image" src="https://github.com/NCAR/geocat-examples/assets/38434768/02b49579-2fc8-4184-848c-76b4c3c569bb">

It looks like neither the python version or the date is actually making it in.

I think we need to explicitly set the date in the github environment, see [this part in xarray's workflow file](https://github.com/pydata/xarray/actions/runs/6309646247/workflow?pr=8229#L66-L68). They also similarly [set the python version](https://github.com/pydata/xarray/actions/runs/6309646247/workflow?pr=8229#L92) in the github environment, though I'm not sure why your `matrix.python-version` didn't work here.

Also, forget what I said about the environment file hash, the caching automatically appends the file hash to the end anyway, so we don't need to specify it.

Reply:
Alright, looks like this is working better now. The OS bit is duplicated in the key, but it's nice to have up front and I don't think it hurts anything.  xarray is set up the same way.   

I think it was just failing back to the default key before. 

I'll try to keep an eye on how this is working for us as well moving forward.

Reply:
> I think it was just failing back to the default key before.

Ahh, that makes sense.